plugins/scaffolder/src/components/fields/EntityNamePicker/validation.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FieldValidation } from '@rjsf/core';
+import { KubernetesValidatorFunctions } from '@backstage/catalog-model';
```

Comment:
I just want to mention that this is only one possible rule set for names. Some orgs change this, and will want it to act accordingly in the frontend as well.

Reply:
Totally! I believe @benjdlambert has some WIP documentation for scaffolder field extensions. Once that lands, we can extend it to document the steps required to configure an EntityNamePicker with different validation rules.